src/stream.iced

```diff
@@ -0,0 +1,56 @@
+stream = require('stream')
+enc = require('./encoding')
+
+desired_high_water_mark = 4096
+calculate_high_water_mark = (input_length) ->
+  input_length*Math.floor(desired_high_water_mark/input_length)
+
+exports.StreamEncoder = class StreamEncoder extends stream.Transform
+
+  constructor : (@encoder) ->
+    @extra = null
+    @block_size = @encoder.in_block_len
+    super({highWaterMark : calculate_high_water_mark(@block_size)})
```

Comment:
It'll work with whatever `highWaterMark` you supply. I left it in because the default is 16k, which seems like a bit much.